src/dev/flang/air/Clazz.java

```diff
@@ -2111,13 +2111,17 @@ Clazz findOuter(AbstractFeature o, HasSourcePosition pos)
     /* starting with feature(), follow outer references
      * until we find o.
      */
-    var res = this;
-    var i = feature();
+    var res = isChoice() ? _outer : this;
+    var i = isChoice() ? _outer.feature() : feature();
     while (i != o && i.outerRef() != null)
       {
         res = res.lookup(i.outerRef(), pos).resultClazz();
         i = i.outer();
       }
+
+    if (POSTCONDITIONS) ensure
+      (i == o);
+
```

Comment:
This should be a `check()` since `i` is an internal local variable. Also, this condition implies that 
```
  while (i != o && i.outerRef() != null)
```
can be simplified to
```
  while (i != o)
```
A good postcondition would be
```
  if (POSTCONDITIONS) ensure
    (res != null); 
```